juanevillam/xtrovert
# Implement Sign-Up Form with Modal/Dialog

## Overview  
This PR implements the sign-up functionality, providing both modal (desktop) and dialog (mobile) forms with `zod` validation and reusable components. It also sets up Redux and `react-hot-toast` for state management and toast notifications.

---

## Key Changes  

### **Features**  
- Set up Redux for state management using `react-redux` and `@reduxjs/toolkit`.  
- Installed `react-hot-toast` for toast notifications.  
- Created modal and dialog components for the sign-up process, both with `zod` validation.  
- Developed reusable components for the sign-up form.  

### **Enhancements**  
- Configured primary and success colors for consistent styling.  

---

## Reasons for Changes  
These changes introduce a comprehensive sign-up feature, enhancing user experience with validation, state management, and notifications. Reusable components improve maintainability, and consistent styling ensures a cohesive UI.

---

## How to Test  
1. Run the application and navigate to the sign-up process.  
2. Verify that the modal (on desktop) and dialog (on mobile) display correctly.  
3. Test input validation by entering valid and invalid data.  
4. Ensure that state management is working correctly and toast notifications appear as expected.  
5. Check that the reusable components function properly in different contexts.
